examples/03_renderable/main.rs

```diff
@@ -68,123 +56,116 @@ impl System<()> for ExampleSystem {
     }
 }
 
-
-fn set_pipeline_state(pipeline: &mut Pipeline, forward: bool) {
+fn set_pipeline_state(pipe: &mut Pipeline, forward: bool) {
     if forward {
-        pipeline.layers = vec![
-            Layer::new("main", vec![
-                Clear::new([0.0, 0.0, 0.0, 1.0]),
-                DrawShaded::new("main", "main"),
-            ]),
-        ];
+        let layer = Layer::new("main",
+                               vec![Clear::new([0.0, 0.0, 0.0, 1.0]),
+                                    DrawShaded::new("main", "main")]);
+        pipe.layers.push(layer);
```

Comment:
This might run into problems, as switching between forward/deferred will just keep on adding layers, instead of replacing the current layers, if I'm not mistaken.